manifests/database_server/firewall.pp

```diff
@@ -0,0 +1,31 @@
+# @summary Configure the firewall settings for OpenAFS database server
+#
+# @param dports
+#   Array of destination ports that need to be open for the OpenAFS database server
+#
+# @param proto
+#   String of protocol that needs to be open for the OpenAFS database server
+#
+# @param sources
+#   Array CIDR sources that need to be open for the OpenAFS database server
+#
+# @example
+#   include openafs::database_server::firewall
+class openafs::database_server::firewall (
+  Array[String] $dports,
+  String        $proto,
+  Array[String] $sources,
+) {
+
+  $sources.each | $location, $source |
+  {
+    firewall { "200 allow OpenAFS via ${proto} from ${source}":
+      proto    => $proto,
+      dport    => $dports,
+      source   => $source,
+      action   => 'accept',
+      provider => 'iptables',
```

Comment:
I don't think this is needed. The point of Puppet is to abstract the details of implementation from the configuration description. The firewall module should be trusted to implement the change properly for the OS.

If there is a known use case in which this is needed, perhaps it should be specified in hiera so it's defined in a single place. Then if an override is deemed necessary,